docs/reference/remap/functions/redact.cue

```diff
@@ -0,0 +1,94 @@
+package metadata
+
+remap: functions: redact: {
+	category: "String"
+	description: """
+		Redact potentially sensitive data in `value` such as:
+
+		- credit card numbers
+		- custom patterns
+		- (more to come!)
+
+		This can act as a last resort in the case your applications log sensitive data by accident to ensure you do not
+		leak this information into your log storage.
+		"""
+
+	arguments: [
+		{
+			name: "value"
+			description: #"""
+				The value to redact sensitive data from.
+
+				Its behavior differs depending on the type of `value`:
+				- For strings, it simply redacts the sensitive data and returns a new string
+				- For arrays, it redacts the sensitive data in each string element
+				- For objects, it masks the sensitive data in each string value, but not keys
+
+				For arrays and objects it will recurse into any nested arrays or objects. Any non-string elements will
+				be skipped.
+
+				Any redacted text will be replaced with `[REDACTED]`.
+				"""#
+			required: true
+			type: ["string", "object", "array"]
+		},
+		{
+			name: "filters"
+			description: #"""
+					List of filters to be applied to the `value`.
+
+					If none, the default set of filters is applied: ["credit_card"].
+
+					Each filter can be specified in one of three ways:
+					- As a regular expression directly, which will be used to redact text matching it
+					- As an object with a `type` key that corresponds to a named filter and additional keys for customizing that filter
+					- As a named filter, if it has no required parameters
+
+					Named filters are:
+					- `pattern`: Redact text matching any regular expressions specified in the, required, `patterns`
+					   key. This is the expanded form of just passing a regular expression as a filter.
+					- `credit_card`: Redact credit card numbers.
```

Comment:
Can we be more specific? I'm curious if there is a pattern standard we can reference here.

Reply:
Ex: https://en.wikipedia.org/wiki/ISO/IEC_7812

Reply:
Yeah, this a good call out. I was a bit hasty adding a credit card filter using a regex I found On The Internet. I decided to drop it from this PR and replace it with a social security card filter which has a simple well-defined format in 5eb934f . I'll leave readding a credit card filter to https://github.com/timberio/vector/issues/5222 where we can take a more thorough approach to it.